### PR TITLE
various bug fixes when integrate test with hdfs store

### PR DESCRIPTION
**Pull request by tuGithub**

### CODE REVIEW

**Comment by tuGithub** [comment]
@chavdar , @sahilTakiar 

please review it when you have time

File: gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStore.java [hunk 1]
```diff
@@ -301,7 +301,11 @@ public Config getOwnConfig(ConfigKeyPath configKey, String version)
    * datasets.
    */
   private Path getDatasetDirForKey(ConfigKeyPath configKey, String version) throws VersionDoesNotExistException {
-    return new Path(getVersionRoot(version), getDatasetFromConfigKey(configKey));
+    String datasetFromConfigKey = getDatasetFromConfigKey(configKey);
+    if(datasetFromConfigKey==null || datasetFromConfigKey.trim().equals(""))
```
Comment (pcadabam) [inline gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStore.java]: StringUtils.isNotBlank

File: gobblin-config-management/gobblin-config-client/src/main/resources/META-INF/services/gobblin.config.store.api.ConfigStoreFactory [hunk 2]
```diff
@@ -0,0 +1 @@
+gobblin.config.store.hdfs.SimpleHDFSConfigStoreFactory
```
Comment (sahilTakiar) [inline gobblin-config-management/gobblin-config-client/src/main/resources/META-INF/services/gobblin.config.store.api.ConfigStoreFactory]: What's this file for?

**Comment by sahilTakiar** [comment]
Pending Pradhan's comment, the changes to the `SimpleHDFSConfigStore` LGTM.

File: gobblin-config-management/gobblin-config-client/src/main/resources/META-INF/services/gobblin.config.store.api.ConfigStoreFactory (hunk 2 above)
Comment (tuGithub) [inline gobblin-config-management/gobblin-config-client/src/main/resources/META-INF/services/gobblin.config.store.api.ConfigStoreFactory]: This is used for ServiceLoader to load service automatically

File: gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStore.java (hunk 1 above)
Comment (tuGithub) [inline gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStore.java]: fixed

**Comment by tuGithub** [comment]
@sahilTakiar 

Please review with the updated version

**Comment by sahilTakiar** [comment]
Everything LGTM, but I don't understand what `getImportsRecursivelyIncludePhantomFromCache` is for can you add a more detailed explanation?

**Comment by tuGithub** [comment]
@sahilTakiar 

Phantom node is the directory which does NOT exist on HDFS. For example, we have /data/databases in the ConfigStore, the phantom node may be /data/databases/foobar

The ConfigClient.getConfig(phantomNode) will return the resolved config of it's parent ( or ancestor )
So, just to make the logic in sync, the ConfigClient.getImports(phantomNode, true) , which return the imports recursively, need to return the imports recursively result of it's parent ( or ancestor )

**Comment by sahilTakiar** [comment]
Why is that only applicable for `getImportsRecursively`?

**Comment by sahilTakiar** [comment]
Discussed offline, LGTM

**Pull request merged**